public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -202,23 +219,34 @@ export const GeneratePopoverBody: React.FC<{
   const renderInnerFooter = () => {
     return (
       <EuiPopoverFooter className="incontextInsightGeneratePopoverFooter" paddingSize="none">
-        <EuiFlexGroup gutterSize="none" direction={'rowReverse'}>
-          {insightAvailable && (
-            <EuiFlexItem
-              grow={false}
-              onClick={() => {
+        {
+          <div style={{ display: showInsight ? 'none' : 'block' }}>
+            <MessageActions
+              contentToCopy={summary}
+              showFeedback
+              showTraceIcon={insightAvailable}
+              onViewTrace={() => {
                 setShowInsight(true);
               }}
-            >
-              <EuiIconTip
-                aria-label="Insight"
-                type="iInCircle"
-                content="Insight with RAG"
-                color={showInsight ? 'blue' : 'black'}
-              />
-            </EuiFlexItem>
-          )}
-        </EuiFlexGroup>
+              usageCollection={usageCollection}
+              isOnTrace={showInsight}
+              metricAppName={metricAppName}
+            />
+          </div>
+        }
+        {
+          <div style={{ display: showInsight && insightAvailable ? 'block' : 'none' }}>
+            <MessageActions
+              contentToCopy={insight}
+              showFeedback
+              showTraceIcon={insightAvailable}
+              onViewTrace={() => {}}
+              usageCollection={usageCollection}
+              isOnTrace={showInsight}
+              metricAppName={metricAppName}
+            />
+          </div>
+        }
       </EuiPopoverFooter>
```

Comment:
```suggestion
        {
          <div style={{ display: showInsight && insightAvailable ? 'block' : 'none' }}>
            <MessageActions
              contentToCopy={insight}
              showFeedback
              showTraceIcon={insightAvailable}
              onViewTrace={() => {}}
              usageCollection={usageCollection}
              isOnTrace={showInsight}
              metricAppName={metricAppName}
            />
          </div>
        }
      </EuiPopoverFooter>
```

{ showInsight && insightAvailable && 
    (<div>
        <MessageActions
            contentToCopy={insight}
            showFeedback
            showTraceIcon={insightAvailable}
            onViewTrace={() => {}}
            usageCollection={usageCollection}
            isOnTrace={showInsight}
            metricAppName={metricAppName}
        />
    </div>)
}

Reply:
we can put the feedbackResult into parent component.